### PR TITLE
Test subprojects build in CI

### PR DESCRIPTION
**Pull request by tristan957**
Since we provide subprojects for various dependencies, let's test to
make sure our subprojects actually work.

Signed-off-by: Tristan Partin <tpartin@micron.com>

### CODE REVIEW

File: .github/workflows/builds.yaml [hunk 1]
```diff
@@ -380,3 +380,51 @@ jobs:
...
+          meson test -C builddir --setup=ci --print-errorlogs --no-stdsplit
+
+      - uses: actions/upload-artifact@v2
+        if: failure()
+        with:
+          name: ${{ github.job }}
+          path: |
+            builddir/meson-logs/
+            /var/log/messages
+            /var/log/syslog
```
Comment (beckerg) [inline .github/workflows/builds.yaml:430]: Would probably be good to collect only the logs generated by the test.  Maybe you can run a command to rotate the log file, or grab the before and after logs and trim the duplicate entries?

**Review by beckerg — APPROVED** [review]

File: .github/workflows/builds.yaml (hunk 1 above)
Comment (tristan957) [reply .github/workflows/builds.yaml:430]: I see what you are saying. You want to truncate /var/log/messages and /var/log/syslog prior to running tests?

Comment (beckerg) [reply .github/workflows/builds.yaml:430]: Well, ideally we wouldn't truncate it, but if this is running on a CI system that just came into existance for this test (or just this session?) then maybe it doesn't matter...

Comment (tristan957) [reply .github/workflows/builds.yaml:430]: It is spun up specifically for this purpose and then deallocated.

**Review by bhaveshvasandani — APPROVED** [review]

**Pull request merged**